loader/include/Geode/ui/TextAreaV2.hpp

```diff
@@ -0,0 +1,241 @@
+#pragma once
+
+#include <cocos2d.h>
+#include "PaddingNode.hpp"
+
+namespace geode {
+    class GEODE_DLL TextAreaV2 : public PaddingNode {
+    public:
+        enum class WrappingMode {
+            // Doesn't wrap the text and completely ignores bounds
+            NoWrap,
+            // Wraps the text on the last special character (spaces included) before the width is exceeded
+            WordWrap,
+            // Wraps the text on the last space before the width is exceeded
+            SpaceWrap,
+            // Wraps the text on the exact character that exceeds the width
+            CutoffWrap
+        };
+
+        enum class Alignment {
+            // Aligns the text to the left
+            Left,
+            // Aligns the text in the center
+            Center,
+            // Aligns the text to the right
+            Right
+        };
+
+        struct Line {
```

Comment:
This should be PIMPL'd, or in some other way not be exposed in a header! Guaranteed ABI break in the future!